### PR TITLE
[#369] Change sw display ordering

### LINKED ISSUE #369: Check how software is ordered in the catalog
– Search results: it should be ordered by relevance and descending vitality
– Software page: it should be ordered randomly
– Home page: it should be ordered randomly
– Public entity page: it should be ordered alphabetically

### PR DESCRIPTION
**Pull request by libremente**
As mentioned in #369, the sw ordering should change.
As such, this PR changes the following:

- [x] Search results: it should be ordered by relevance and descending vitality
- [x] Home page: it should be ordered randomly
- [x] Public entity page: it should be ordered alphabetically
-  ~Software page: it should be ordered randomly~

Fix:
* Fixes #369

### CODE REVIEW

**Comment by libremente** [comment]
@sebbalex I need an hint on the first point. I'd probably work on this https://github.com/italia/developers.italia.it/blob/8ca39de47287c3372838b979bc8f859c650e86b3/assets/scripts/esDevelopersItaliaQuery.js#L173-L177 but you are more familiar with that codebase. Suggestions? We need to default at relevance + descending vitality

File: _layouts/brand-pa.html
Comment (libremente) [inline _layouts/brand-pa.html:16]:
```suggestion
{% assign generic_sw = site.data.crawler.softwares | where_exp: "item", "item.publiccode.it.riuso.codiceIPA == page.ipa"  | upcase | sort: "publiccode.name" | slice: 0, 60 %}
```

File: _layouts/brand-pa.html
Comment (libremente) [inline _layouts/brand-pa.html:16]:
```suggestion
{% assign generic_sw = site.data.crawler.softwares | where_exp: "item", "item.publiccode.it.riuso.codiceIPA == page.ipa"  | sort_natural: "publiccode.name" | slice: 0, 60 %}
```

**Comment by libremente** [comment]
I have still issues with `sort_natural`, I am unsure whether it's the case of adding an external plugin just to handle the sorting, thoughts @sebbalex?

**Comment by sebbalex** [comment]
> I have still issues with `sort_natural` since it's not working as it should, I am unsure whether it's the case of adding an external plugin just to handle the sorting, thoughts @sebbalex?

Ok, I dig in a bit and I found out which is the root cause: `sort_natural` doesn't support nested field. See here for details: https://github.com/Shopify/liquid/pull/604
I tried filtering with root attribute but with no luck, they are too inconsistent. What do you suggest? cc @libremente

**Comment by immuniopensource** [comment]
> > I have still issues with `sort_natural` since it's not working as it should, I am unsure whether it's the case of adding an external plugin just to handle the sorting, thoughts @sebbalex?
> 
> Ok, I dig in a bit and I found out which is the root cause: `sort_natural` doesn't support nested field. See here for details: [Shopify/liquid#604](https://github.com/Shopify/liquid/pull/604)
> I tried filtering with root attribute but with no luck, they are too inconsistent. What do you suggest? cc @libremente

Thanks! I guess the solution is something like this https://github.com/johnnyshields/naturalsort

**Comment by libremente** [comment]
> > I have still issues with `sort_natural` since it's not working as it should, I am unsure whether it's the case of adding an external plugin just to handle the sorting, thoughts @sebbalex?
> 
> Ok, I dig in a bit and I found out which is the root cause: `sort_natural` doesn't support nested field. See here for details: [Shopify/liquid#604](https://github.com/Shopify/liquid/pull/604)
> I tried filtering with root attribute but with no luck, they are too inconsistent. What do you suggest? cc @libremente

@sebbalex I'd suggest to write a simple rb plugin to make the sort_natural work. I'm sure we can use it also for other things.

**Comment by bfabio** [comment]
We should squash these commits once this is ready to be merged.

**Comment by bfabio** [comment]
@libremente WDYT?

**Comment by libremente** [comment]
@bfabio yes definitely, can you go ahead and squash as you like? Thanks

**Comment by libremente** [comment]
As far as I can tell it's working like a charm!!! :rocket: great job @bfabio!! Let's wait for @sebbalex review then we can go ahead an merge.

**Review by bfabio — APPROVED** [review]
lgtm :man_shrugging:

**Comment by bfabio** [comment]
@libremente Done. I can't add you as a reviewer because you're the author of the PR :)

**Review by sebbalex — APPROVED** [review]
great job! :rocket:

**Comment by libremente** [comment]
Great, merging! :+1:

**Pull request merged**
